### PR TITLE
Add validation annotations.

### PR DESCRIPTION
**Pull request by tjholm**
validation rules are not final and are examples only, for full docs on options see:
https://github.com/envoyproxy/protoc-gen-validate

Need to determine proper regexes for id/resource names.

### CODE REVIEW

**Comment by medgar-nitric** [comment]
Looks great. These message layouts are bit different from what we are currently doing, but happy to go with these. Given the way the messages will be wrapped as GRPC errors with a service/method name it will be pretty clear what is wrong.

In the future we may want to think about localization of error messages, but that is a long long way down the track after we achieve world domination.

**Comment by tjholm** [comment]
Only thing blocking this is consensus on regex patterns for naming.

**Review by medgar-nitric — APPROVED** [review]
This PR looks good to me. I am happy with the regex conventions. I think we can get feedback from customers.  The maximum length seems very long to me, however my practical experience with this is think of the longest name you could imagine, and then double it.

We should raise a ticket on the docs project, to update the various services with these conventions.

File: proto/secret/v1/secret.proto [hunk 1]
```diff
@@ -34,28 +36,31 @@ message SecretPutResponse {
...
+  SecretVersion secret_version = 1 [(validate.rules).message.required = true];
 }
 
 // The secret response
 message SecretAccessResponse {
   // The version of the secret that was requested
-  SecretVersion secret_version = 1;
+  SecretVersion secret_version = 1 [(validate.rules).message.required = true];
   // The value of the secret
   bytes value = 2;
```
[the remark this answers is not in the document]
Comment (tjholm) [reply proto/secret/v1/secret.proto:47]: well spotted, will pop that in :)

File: proto/document/v1/document.proto [hunk 2]
```diff
@@ -94,23 +104,23 @@ message DocumentGetResponse {
 
 message DocumentSetRequest {
   // Key of the document to set
-  Key key = 1;
+  Key key = 1 [(validate.rules).message.required = true];
   // The document content to store (JSON object)
-  google.protobuf.Struct content = 3;
+  google.protobuf.Struct content = 3 [(validate.rules).message.required = true];
```
[the remark this answers is not in the document]
Comment (tjholm) [reply proto/document/v1/document.proto:109]: Unfortunately not from the looks of it: https://github.com/envoyproxy/protoc-gen-validate

only message types are valid for structs so we can only make sure that they actually set it.

We'll have to implement it in the membrane in this case.

File: proto/secret/v1/secret.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (tjholm) [reply proto/secret/v1/secret.proto]: done

**Review by medgar-nitric — APPROVED** [review]

**Review by jyecusch — APPROVED** [review]

**Pull request merged**